src/PowerShellEditorServices/Services/DebugAdapter/Handlers/StackTraceHandler.cs

```diff
@@ -1,64 +1,132 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
+#nullable enable
 
 using System;
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Management.Automation;
 using Microsoft.PowerShell.EditorServices.Services;
-using Microsoft.PowerShell.EditorServices.Services.DebugAdapter;
-using Microsoft.PowerShell.EditorServices.Utility;
 using OmniSharp.Extensions.DebugAdapter.Protocol.Models;
 using OmniSharp.Extensions.DebugAdapter.Protocol.Requests;
+using Microsoft.PowerShell.EditorServices.Services.DebugAdapter;
+using System.Linq;
+
+namespace Microsoft.PowerShell.EditorServices.Handlers;
 
-namespace Microsoft.PowerShell.EditorServices.Handlers
+internal class StackTraceHandler(DebugService debugService) : IStackTraceHandler
 {
-    internal class StackTraceHandler : IStackTraceHandler
+    /// <summary>
+    /// Because we don't know the size of the stacktrace beforehand, we will tell the client that there are more frames available, this is effectively a paging size, as the client should request this many frames after the first one.
+    /// </summary>
+    private const int INITIAL_PAGE_SIZE = 20;
+
+    public async Task<StackTraceResponse> Handle(StackTraceArguments request, CancellationToken cancellationToken)
     {
-        private readonly DebugService _debugService;
+        if (!debugService.IsDebuggerStopped)
+        {
+            throw new NotSupportedException("Stacktrace was requested while we are not stopped at a breakpoint.");
```

Comment:
Do we know for sure this is caught somewhere? I want to make sure we don't crash if our assumptions are incorrect and this can happen.

Reply:
I will verify.

Reply:
I changed it to RpcErrorException so that it will only kill the handler and log an error to PSES. I have a better Exception for this in the rename PR that eventually we will backport all these RpcErrorExceptions to.